tracks/ruby/exercises/high-scores/mentoring.md

```diff
@@ -0,0 +1,71 @@
+New concepts: Array query methods (+ chaining methods), instantiating a class, getter method or `attr_reader`.
+
+### Minimal solution for approval
+
+```ruby
+
+# version 2+
+
+class HighScores
+  def initialize(scores)
+    @scores = scores
+  end
+
+  def scores  # no need to change getter method to `attr_reader`
+    @scores
+  end
+
+  def personal_best
+    scores.max
+  end
+
+  def latest
+    scores.last
+  end
+
+  def personal_top
+    scores.sort.reverse.take(3) 
+  end
+
+  # composing a string either with string interpolation or with an if.. else.. statement
+  def report
+    difference = "#{personal_best - latest} short of" unless personal_best == latest
+
+    "Your latest score was #{latest}. That's #{difference} your personal best!".squeeze
+  end
+end
+
+```
+
+### Reasonable approaches
+
+- See the one above
+- Defining `@scores`, `@personal_best` and `personal_top` in the constructor.  
+
+### General notes
+- The Instructions point to a beginner friendly explanation of instantiating a class: 
+http://ruby-for-beginners.rubymonstas.org/writing_classes/initializers.html
+and to the rubydocs Array. 
+- Most submissions use an getter method `scores`, not `attr_reader`. 
+That's okay at this point in the track, no need to suggest `attr_reader`.
+- Beginners create the attribute reader, but may call the instance variable instead of the getter. 
+This link gives a nice introduction of getters and setters, and the naming convention in Ruby:
+https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#accessormethods.
+It's recommended to link to this or your favorite explanation, instead of typing the solution for them.
+- `sort.reverse` is not very performant, but at this point in the track it's more important that students explore the 
+standard lib (and work with methods chaining) than that they know how to optimize.
+
+### Talking points
+- `Array#max` takes an argument; `scores.max(3)` works.
+- Most submissions retrieve the `personal_top` values with the `Array#[]` method. 
+`Array#take` wins in readability. And `Array#max()` wins them all. 
+- The `report` method asks for a conditionally composed string. It's meant to think about the trade-off 'duplication' vs 'readability'. Recommendation: approve any solution unless it's a supercomplicated `if/else statement`. 
+- Consider `String#squeeze` to avoid hard-to-spot spacing in the string itself. 
+
+### Mentor Research
+- `String#<< / String.concat` vs. `String#+`? 
```

Comment:
These should be links to the docs. And these notes are pretty meaningless if they don't include what the resolution of the research was.

Reply:
I don't agree about the links to the documentation.  For example, I know that I can do `ri 'String#+'` to get to the documentation, and a link is not going to change this.  Others may not.  Plus the links may decay over time, due to our supported language, or their dynamic nature depending on where the links direct.  I am also aware of students using different versions of Ruby, and so they would have to adjust for their own use anyway.

Reply:
When I do `ri String#+` it gives me

```
$ ri String#+
Nothing known about String
```,

[String#+](https://ruby-doc.org/core-2.5.0/String.html#method-i-2B) however works, regardless the version. In general ruby doesn't _remove_ methods, _ever_, so usually links to older versions are not a problem.

Might just be me though :)

Reply:
You probably haven't generated the ri documentation for core.  But your gems will show.  Once you generate the core documentation, it will show, if the documentation for gems are showing.